test cases/common/98 gen extra/meson.build

```diff
@@ -11,3 +11,22 @@ g2 = gen.process('name.dat', extra_args: '--upper')
 
 test('basic', executable('basic', 'plain.c', g1))
 test('upper', executable('upper', 'upper.c', g2))
+
+cp = find_program('cp')
+basename_gen = generator(cp,
+  output : '@BASENAME@.c',
+  arguments : ['@INPUT@', join_paths('@BUILD_DIR@', '@BASENAME@.c')])
```

Comment:
This is identical to just having `@OUTPUT@` there. We don't want to advocate this sort of behavior so it should not be in our tests either. Also, you should not be using the `cp` command because it is not portable. Create a custom generator script instead which you call like this:

    arguments : ['@INPUT@', '@BUILD_DIR@' '@BASENAME@.c']

and which then joins arguments 2 and 3. This is how all other tests dealing with file copying/managing are implemented.

The same for the other test below. Thanks.